src/core/src/device/dummy.rs

```diff
@@ -0,0 +1,50 @@
+#![allow(missing_docs)]
+use device::{Device, Resources, Capabilities, SubmitInfo};
+use device::command::{GenericCommandBuffer};
+
+pub struct DummyDevice {
+    capabilities: Capabilities
+}
+
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DummyResources{}
+
+pub type Buffer         = u32;
```

Comment:
I must have missed the Buffer, ArrayBuffer, and other types after refactoring as I copied it from the gl implementation. Is there a better type than `u32` for dummy device's resources?